### PR TITLE
Add bindings for LocalMessageDefinitionSource

### PR DESCRIPTION
**Pull request by methylDragon**
# Description
This adds bindings for `LocalMessageDefinitionSource.get_full_text` to `rosbag2_py`. I don't think it's too controversial.

Unit tests added and passed.

### CODE REVIEW

File: rosbag2_py/rosbag2_py/_message_definitions.pyi [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from typing import Any
+
+class LocalMessageDefinitionSource:
+    def __init__(self) -> None: ...
+    def get_full_text(self, arg0: str) -> Any: ...
```
Comment (mjcarroll) [inline rosbag2_py/rosbag2_py/_message_definitions.pyi]: The linter doesn't like this.

**Review by fujitatomoya — APPROVED** [review]
lgtm with https://github.com/ros2/rosbag2/pull/1697/files#r1631359697 fixed.

File: rosbag2_py/rosbag2_py/_message_definitions.pyi (hunk 1 above)
Comment (methylDragon) [reply rosbag2_py/rosbag2_py/_message_definitions.pyi]: I wonder if the pyi files are generated automatically 🤔 

No matter: https://github.com/ros2/rosbag2/pull/1697/commits/0b6d099f71ca618d002279ddf20c7c79b9bac97f

File: rosbag2_py/src/rosbag2_py/_message_definitions.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2024 Intrinsic Innovcation LLC. All rights reserved.
```
Comment (MichaelOrlov) [inline rosbag2_py/src/rosbag2_py/_message_definitions.cpp]: Typo. Should be "Innovation"
```suggestion
// Copyright 2024 Intrinsic Innovation LLC. All rights reserved.
```

**Comment by methylDragon** [comment]
Looks like I still need reviews from @MichaelOrlov / @james-rms

**Review by MichaelOrlov — COMMENTED** [review]
@methylDragon Thanks for the PR. Overall looks good to me among one minor finding with a typo.

As regards the changes in the `.pyi` files, Yes, they are supposed to be autogenerated. Please see the instruction in the readme file in the `rosbag2_py` package https://github.com/ros2/rosbag2/blob/rolling/rosbag2_py/README.md

File: rosbag2_py/src/rosbag2_py/_message_definitions.cpp (hunk 2 above)
Comment (methylDragon) [reply rosbag2_py/src/rosbag2_py/_message_definitions.cpp]: Oops, thanks!

**Review by MichaelOrlov — APPROVED** [review]
LGTM! with green CI.

**Comment by MichaelOrlov** [comment]
Pulls: ros2/rosbag2#1697
Gist: https://gist.githubusercontent.com/MichaelOrlov/e971eb33cd4164b7b2dcb9cb092fe7b9/raw/dc445ec0c9644a863e7458f6b19133245acade30/ros2.repos
BUILD args:  --packages-above-and-dependencies rosbag2_py rosbag2_tests ros2bag
TEST args:  --packages-above rosbag2_py rosbag2_tests ros2bag
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14049
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21239/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15557/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/716/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21961/)

**Comment by MichaelOrlov** [comment]
https://github.com/Mergifyio backport jazzy

**Pull request merged**